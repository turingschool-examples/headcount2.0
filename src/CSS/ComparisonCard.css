.compare-data-card {
  animation: shazam 1s linear, bazinga 1s ease-in;
  animation: shazam 1s linear, bazinga 1s ease-out;
  grid-column: 2 / 3;
  grid-row: 1 / 2;
  justify-self: center;
  align-self: center;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  width: 38rem;
  text-align: center;
}

.district-avg {
  margin: 3rem 0;
}

.district-low {
  color: rgb(169, 77, 77);
  font-size: 2rem;
}

.district-high {
  color: green;
  font-size: 2rem;
}

.comparison-value {
  animation: morphingLogo 4s linear infinite;
  color: rgb(74, 96, 125);
  font-size: 4.5rem;
  margin: 4rem auto 0 auto;
}

@keyframes morphingLogo {
  0% {
    filter: hue-rotate(-90deg);
  }
  50% {
    filter: hue-rotate(90deg);
  }
  100% {
    filter: hue-rotate(-90deg);
  }
}

@keyframes shazam {
  0% {
    right: 250px;
  }
  100% {
    right: 10px;
  }
}

@keyframes bazinga {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
