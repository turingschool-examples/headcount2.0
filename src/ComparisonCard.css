.comparison-card {
  background-color: #9b9b9b;
  border: 8px solid black;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 270px;
  justify-content: space-evenly;
  margin: 10px;
  opacity: 0.9;
  padding-bottom: 10px;
  text-align: center;
  width: 280px;
}

.comparison-card h3 {
  font-size: 20px;
  font-weight: bold;
  line-height: 2.8;
  margin-bottom: 0;
  margin-top: 0;
  padding: 0;
  position: relative;
}

.comparison-card p {
  background-color: lightgray;
  border: 2px solid black;
  border-radius: 20px;
  font-size: 16px;
  font-weight: bold;
  margin: auto;
  padding: 4px;
  width: 60px;
}

.avg-bottom {
  margin-left: 40px;
}

.avg-top {
  margin-right: 40px;
}

.pencil-icon {
  height: 42px;
  margin-right: 26px;
  position: relative;
  top: 13px;
  transform: rotate(45deg);
}

.rotate {
  margin-left: 26px;
  transform: rotate(225deg);
}

.apple-placeholder {
  background-color: #9b9b9b;
  border: 8px solid black;
  border-radius: 20px;
  height: 280px;
  margin: 10px;
  opacity: 0.9;
  width: 280px;
}

.apple-pic {
  display: flex;
  height: 150px;
  margin: 55px auto 0 auto;
}

@media screen and (max-width: 905px) {
  .pencil-icon {
    transform: rotate(135deg);
  }

  .rotate {
    transform: rotate(315deg);
  }
}
