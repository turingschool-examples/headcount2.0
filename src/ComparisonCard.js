import React from "react";

export const ComparisonCard = ({ comparisonData }) => {
  const comparisonKeys = Object.keys(comparisonData);
  const districtKeyOne = comparisonKeys[0];
  const districtKeyTwo = comparisonKeys[1];
  return (
    <div>
      <p>
        {comparisonKeys[0]} :<span> {comparisonData[districtKeyOne]}</span>
      </p>
      <container>
        ----
        {comparisonData[comparisonKeys[2]]}
        ----
      </container>
      <p>
        {comparisonKeys[1]} :<span> {comparisonData[districtKeyTwo]}</span>
      </p>
    </div>
  );
};
