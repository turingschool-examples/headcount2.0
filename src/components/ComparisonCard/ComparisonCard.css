.compare-card h1 {
  font-size: 3em;
  margin: 0;
}

.avg {
  font-size: 1.5em;
  color: #78C9D0 !important;
}

.bottom {
  margin-bottom: 0;
}

.name-target {
  margin: 35px 0 10px 0;
}