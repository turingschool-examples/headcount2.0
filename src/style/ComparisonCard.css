.comparison-card {
  margin: 0 auto;
  background-color: rgb(232, 232, 232);
  width: 540px;
  padding: 20px;
  color: #54616D;
  border: 1px solid black;
  box-shadow: 10px 10px 30px -12px rgba(0,0,0,0.75);
  margin-bottom: 20px;
}

.district-info {
  display: flex;
  justify-content: space-between;
  font-size: 1.3rem;
}

.comparison-ratio {
  text-align: center;
  font-size: 1.5rem;

}