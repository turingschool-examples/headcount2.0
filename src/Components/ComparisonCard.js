import React from "react";
import PropTypes from "prop-types";
import "../CSS/ComparisonCard.css";

export const ComparisonCard = ({ comparisonData }) => {
  const comparisonKeys = Object.keys(comparisonData);
  const districtKeyOne = comparisonKeys[0];
  const districtKeyTwo = comparisonKeys[1];
  return (
    <div className="compare-data-card">
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] < comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyOne]}
        </p>
      </div>
      <p className="comparison-value">{comparisonData[comparisonKeys[2]]}</p>
      <div className="district-title">
        Aggregate Avg.
        <p
          className={
            comparisonData[districtKeyOne] > comparisonData[districtKeyTwo]
              ? "district-low"
              : "district-high"
          }
        >
          {comparisonData[districtKeyTwo]}
        </p>
      </div>
    </div>
  );
};

const { shape, number } = PropTypes;

ComparisonCard.propTypes = {
  comparisonData: shape({ year: number })
};
