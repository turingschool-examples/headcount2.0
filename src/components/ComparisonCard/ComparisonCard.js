import React from 'react';
import './ComparisonCard.css';
import PropTypes from 'prop-types';

const ComparisonCard = ({comparedObject}) => {
  if (comparedObject) {
    const d1Name = Object.keys(comparedObject)[0];
    const d1Avg = comparedObject[Object.keys(comparedObject)[0]];
    const d2Name = Object.keys(comparedObject)[1];
    const d2Avg = comparedObject[Object.keys(comparedObject)[1]];
    const compareRatio = comparedObject.compared;
    
    return (
      <article className="compare-card">
        <h3>{d1Name}</h3>
        <h4 className="avg">{d1Avg}</h4>
        <h1>{compareRatio}</h1>
        <h4 className="avg bottom">{d2Avg}</h4>
        <h3 className="name-target">{d2Name}</h3>
      </article>
    );
  }
  
  return (
    <div></div>
  );
};

ComparisonCard.propTypes = {
  comparedObject: PropTypes.object
};

export default ComparisonCard;