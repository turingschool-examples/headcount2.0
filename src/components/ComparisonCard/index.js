import React from "react";
import applePic from "../../images/apple.png";
import pencilIcon from "../../images/pencil.svg";
import PropTypes from "prop-types";
import "./ComparisonCard.css";

const ComparisonCard = props => {
  const districtNames = Object.keys(props);
  if (districtNames[0] === "props" || districtNames[1] === "props") {
    return (
      <div className="apple-placeholder">
        <img className="apple-pic" alt="apple" src={applePic} />
      </div>
    );
  } else {
    return (
      <div className="comparison-card">
        <h3 className="avg-top">
          <img className="pencil-icon" alt="pencil" src={pencilIcon} />
          AVERAGE
        </h3>
        <p> {props[districtNames[0]]}</p>
        <h3>COMPARED AVERAGE</h3>
        <p> {props.compared}</p>
        <h3 className="avg-bottom">
          AVERAGE
          <img className="pencil-icon rotate" alt="pencil" src={pencilIcon} />
        </h3>
        <p> {props[districtNames[1]]}</p>
      </div>
    );
  }
};

ComparisonCard.defaultProps = {
  props: PropTypes.obj
};

ComparisonCard.propTypes = {
  compared: PropTypes.number
};

export default ComparisonCard;
