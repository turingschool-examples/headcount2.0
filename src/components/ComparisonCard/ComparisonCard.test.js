/* eslint-disable */
import React from 'react';
import ComparisonCard from './ComparisonCard';
import { shallow } from 'enzyme';

describe('ComparisonCard', () => {

  it('snapshot test', () => {
    const mockObject = {'ADAMS COUNTY 14': 0.709, 'ACADEMY 20': 0.407, compared: 1.742};
    const wrapper = shallow(<ComparisonCard 
                              comparedObject={mockObject}/>);
    expect(wrapper).toMatchSnapshot();
  });
});