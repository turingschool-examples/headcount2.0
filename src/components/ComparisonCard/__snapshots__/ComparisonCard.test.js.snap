// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    comparedObject={
        Object {
            "ACADEMY 20": 0.407,
            "ADAMS COUNTY 14": 0.709,
            "compared": 1.742,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          ADAMS COUNTY 14
</h3>,
        <h4
          className="avg"
>
          0.709
</h4>,
        <h1>
          1.742
</h1>,
        <h4
          className="avg bottom"
>
          0.407
</h4>,
        <h3
          className="name-target"
>
          ACADEMY 20
</h3>,
      ],
      "className": "compare-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADAMS COUNTY 14",
        },
        "ref": null,
        "rendered": "ADAMS COUNTY 14",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.709,
          "className": "avg",
        },
        "ref": null,
        "rendered": 0.709,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1.742,
        },
        "ref": null,
        "rendered": 1.742,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0.407,
          "className": "avg bottom",
        },
        "ref": null,
        "rendered": 0.407,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ACADEMY 20",
          "className": "name-target",
        },
        "ref": null,
        "rendered": "ACADEMY 20",
        "type": "h3",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            ADAMS COUNTY 14
</h3>,
          <h4
            className="avg"
>
            0.709
</h4>,
          <h1>
            1.742
</h1>,
          <h4
            className="avg bottom"
>
            0.407
</h4>,
          <h3
            className="name-target"
>
            ACADEMY 20
</h3>,
        ],
        "className": "compare-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADAMS COUNTY 14",
          },
          "ref": null,
          "rendered": "ADAMS COUNTY 14",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.709,
            "className": "avg",
          },
          "ref": null,
          "rendered": 0.709,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1.742,
          },
          "ref": null,
          "rendered": 1.742,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0.407,
            "className": "avg bottom",
          },
          "ref": null,
          "rendered": 0.407,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ACADEMY 20",
            "className": "name-target",
          },
          "ref": null,
          "rendered": "ACADEMY 20",
          "type": "h3",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
